HGCalAnalysis/plugins/HGCalAnalysis.cc

```diff
@@ -448,622 +510,683 @@ HGCalAnalysis::~HGCalAnalysis()
+
+        if (!it_track->quality(reco::Track::highPurity)) continue;
+
+        double energy = it_track->pt() * cosh(it_track->eta());
+
+        //save info about reconstructed tracks propoagation to hgcal layers (ony for pt>propagationPtThreshold tracks)
+        std::vector<float> xp,yp,zp;
+
+        if (it_track->pt()>= propagationPtThreshold) {
+
+            // Define error matrix
+            ROOT::Math::SMatrixIdentity id;
+            AlgebraicSymMatrix55 C(id);
+            C *= 0.01;
+            CurvilinearTrajectoryError err(C);
+            typedef TrajectoryStateOnSurface TSOS;
+
+            GlobalPoint startingPosition(it_track->vx(),it_track->vy(),it_track->vz());
+            GlobalVector startingMomentum(it_track->px(),it_track->py(),it_track->pz());
+
+            Plane::PlanePointer startingPlane = Plane::build( Plane::PositionType (it_track->vx(),it_track->vy(),it_track->vz()), Plane::RotationType () );
+
+            TSOS startingStateP(GlobalTrajectoryParameters(startingPosition,startingMomentum, it_track->charge(), aField), err, *startingPlane);
+
+            for(unsigned il=0; il<layerPositions.size(); ++il) {
+                float xp_curr=0;
+                float yp_curr=0;
+                float zp_curr=0;
+
+                for (int zside = -1; zside <=1; zside+=2) {
+                    // clearly try both sides
+                    Plane::PlanePointer endPlane = Plane::build( Plane::PositionType (0,0,zside*layerPositions[il]), Plane::RotationType());
+                    try{
+                        /*
+                          std::cout << "Trying from " <<
+                          " layer " << il <<
+                          " starting point " << startingStateP.globalPosition() <<
+                          std::endl;
+                        */
+                        TSOS trackStateP = RKProp.propagate( startingStateP, *endPlane);
+                        if (trackStateP.isValid()) {
+                            xp_curr=trackStateP.globalPosition().x();
+                            yp_curr=trackStateP.globalPosition().y();
+                            zp_curr=trackStateP.globalPosition().z();
+
+                            // std::cout << "Succesfully finished Positive track propagation  -------------- with RK: " << trackStateP.globalPosition() << std::endl;
+                        }
+                    }
+                    catch (...) {
+                        std::cout << "MagVolumeOutsideValidity not properly caught!! Lost this track " << std::endl;
+                    }
+                }
+                xp.push_back(xp_curr);
+                yp.push_back(yp_curr);
+                zp.push_back(zp_curr);
+            } // closes loop on layers
+        } // closes conditions pt>3
+
+        // save info in tree
+        track_pt.push_back(it_track->pt());
+        track_pt.push_back(it_track->eta());
```

Comment:
I don't know how this has happened, but once I've skipped my own indentation commit, it was fixed again